lib/wasi/src/lib.rs

```diff
@@ -116,12 +149,77 @@ impl WasiEnv {
             { Box::new(()) as Box<dyn NamedResolver + Send + Sync> };
         for version in wasi_versions.iter() {
             let new_import_object =
-                generate_import_object_from_env(module.store(), self.clone(), *version);
+                generate_import_object_from_thread(module.store(), self.clone(), *version);
             resolver = Box::new(new_import_object.chain_front(resolver));
         }
         Ok(resolver)
     }
 
+    /// Get a reference to the memory
+    pub fn memory(&self) -> &Memory {
+        self.memory_ref()
+            .expect("Memory should be set on `WasiThread` first")
+    }
+
+    // Copy the lazy reference so that when its initialized during the
+    // export phase that all the other references get a copy of it
```

Comment:
```suggestion
    /// Copy the lazy reference so that when its initialized during the
    /// export phase that all the other references get a copy of it
```